DFE-Digital/sts-plan-technology-for-your-school
# feat: Add process for automated contentful backups

## Overview

Addresses ticket [#228964](https://dfe-ssp.visualstudio.com/s190-schools-technology-services/_workitems/edit/228964)

Adds an automated pipeline for backing up Contentful content

## Changes

- Adds a basic storage account + container for contentful backups in each environment
- Adds a pipeline that runs nightly (could probably be less frequent) which backs up content to that container (only for prod / note that for testing the pipeline says 'Dev' but it will be 'Prod')
- Fixes an existing issue with validateEnvironment in the content import script that was introduced by the last migration but not noticed
- Merged the contentful data clearer into the content-management project - it has the same dependencies, and this way it can be called by the import-content script and clear out the environment before repopulating it.

## How to review the PR

- Run the pipeline for Dev and check in Azure that the export file has shown up in the blob container
- Run the content-management import command
  1. First use the export-processor for prod or dev to get a json file
  3. Setup .env variables **make sure to use the DevSandbox**, (leave `DELETE_ALL_DATA` as false)
  4. Run `node import-content`
  5. Check that everything in the import is now in the sandbox environment
- Run it again with data deletion
  1. Add some data to the sandbox (or use whichever env you didn't use in step 1)
  2. Set `DELETE_ALL_DATA=true`
  3. Run `node import-content` again
  4. Check that everything you added is gone, and only the data from the import is in the sandbox
- Separately, test that running `node delete-all-data` clears out all data from the sandbox environment

Add any thoughts

## Requirements for Release
- Add the new KV secrets to test/staging/prod
- Run the tf on staging/prod

## Checklist

- [x] Terraform has been updated
